rpc/src/endpoint/broadcast/tx_commit.rs

```diff
@@ -34,22 +33,34 @@ impl RequestMessage for Request {
     }
 }
 
-impl<S: Dialect> crate::Request<S> for Request {
-    type Response = DialectResponse<S::Event>;
+impl crate::Request<dialect::v0_34::Dialect> for Request {
+    type Response = self::v0_34::DialectResponse;
 }
 
-impl<S: Dialect> crate::SimpleRequest<S> for Request {
+impl crate::Request<dialect::v0_37::Dialect> for Request {
+    type Response = Response;
+}
+
+impl<S: Dialect> crate::SimpleRequest<S> for Request
+where
+    Self: crate::Request<S>,
+    Response: From<Self::Response>,
+{
     type Output = Response;
 }
 
 /// Response from `/broadcast_tx_commit`.
-#[derive(Clone, Debug, Serialize)]
+#[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct Response {
     /// `CheckTx` result
     pub check_tx: abci::response::CheckTx,
 
-    /// `DeliverTx` result
-    pub deliver_tx: abci::response::DeliverTx,
+    /// Result of executing the transaction.
+    ///
+    /// The JSON field carrying this data is named `deliver_tx` in
+    /// CometBFT versions before 0.38.
+    #[serde(alias = "deliver_tx")]
```

Comment:
But we do derive `Serialize` on the struct, and it will serialize to the latest RPC schema, 0.38.
The `DialectResponse` from the submodule can be used to serialize in the 0.34 schema.

And thanks for bringing this up: this means we have nothing to serialize with for 0.37 JSON-RPC if anyone wants to. The RPC client does not use `Serialize`, so I've treated it as an afterthought. Maybe there should be another compatibility submodule just to enable this.